event_response_tool/ap_creator/create_ap.py

```diff
@@ -0,0 +1,237 @@
+                    f"Event Catalog {catalog.id} was uploaded, but failed while processing. {catalog.status_message}"
+                )
+            return catalog
+
+    def upload_simulation(
+        self, event_catalogs, start_date, trial_count, simulation_data
+    ):
+        try:
+            simulation = analyzere.Simulation(
+                name=self.simulation_description,
+                description=self.simulation_description,
+                event_catalogs=event_catalogs,
+                start_date=start_date,
+                trial_count=trial_count,
+            )
+            simulation.save()
+            simulation.upload_data(simulation_data)
+            check_resource_upload_status(simulation)
+        except Exception as e:
+            alert.exception(
+                f"Exception occured while uploading simulation: {e}"
+            )
+        else:
+            if simulation.status == "processing_succeeded":
+                alert.info(
+                    f"Simulation {simulation.id} has been created successfully"
+                )
+            else:
+                alert.error(
+                    f"Simulation {simulation.id} was uploaded, but failed while processing. {simulation.status_message}"
+                )
+            return simulation
+
+    def build_simulation_data(self):
+        alert.debug("Building simulation data")
+        event_column = find_column(
+            "event", self.event_weights_df.columns.tolist()
+        )
+        weight_column = find_column(
+            "weight", self.event_weights_df.columns.tolist()
+        )
+
+        events = self.event_weights_df[event_column].to_list()
+        weights = self.event_weights_df[weight_column].to_list()
+        max_trials = self.trial_count
+        trial_count_by_event = [ceil(max_trials * w) for w in weights]
+        trial_id = 1
+
+        simulation_data = """TrialId,EventId,Day\n"""
+
+        for event_id, trial_count in zip(events, trial_count_by_event):
+            if event_id < self.total_num_of_events:
+                for sample in range(0, trial_count):
+                    simulation_data += "%s,%s,%s\n" % (trial_id, event_id, 1)
+                    trial_id += 1
+
+        return simulation_data, sum(trial_count_by_event)
+
+    def create_simulation(self, catalogs, start_date):
+        simulation_data, trial_count = self.build_simulation_data()
+        alert.debug("Simulation data built successfully")
+
+        simulation = self.upload_simulation(
+            event_catalogs=catalogs,
+            start_date=start_date,
+            trial_count=trial_count,
+            simulation_data=simulation_data,
+        )
+
+        return simulation
+
+    def create_loss_filters(self):
+        loss_filters = [
+            analyzere.LossFilter(
+                type="AnyOfFilter",
+                name="Event %s" % (i),
+                description="Event %s" % (i),
+                attribute="Sequence",
```

Comment:
Can we filter on EventId directly, and avoid the need for `Sequence` (which could be confused for `Day`)?